src/main/java/frc/robot/subsystems/Vision/IOPhotonVision.java

```diff
@@ -0,0 +1,84 @@
+import org.photonvision.targeting.PhotonTrackedTarget;
+
+import java.io.IOException;
+
+public class IOPhotonVision implements VisionIO {
+
+    private final PhotonCamera photonCamera;
+    private final PhotonPipelineResult result;
+    private final PhotonTrackedTarget target;
+    private final AprilTagFieldLayout fieldLayout;
+    private final PhotonPoseEstimator photonPoseEstimator;
+
+    public IOPhotonVision(PhotonCamera camera, int camIndex) {
+        this.photonCamera = camera;
+        result = photonCamera.getLatestResult();
+        target = result.getBestTarget();
+        try {
+            fieldLayout = AprilTagFieldLayout.loadFromResource(AprilTagFields.k2023ChargedUp.m_resourceFile);
+            photonPoseEstimator = new PhotonPoseEstimator(fieldLayout, PhotonPoseEstimator.PoseStrategy.LOWEST_AMBIGUITY,  //TODO: check strategies
+                    photonCamera, VisionConstants.ROBOT_TO_CAM[camIndex]);
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public Pose3d getEstimatedPoseTargetOriented() {
+        var estimatedPose = photonPoseEstimator.update();
+        return estimatedPose.map(estimatedRobotPose -> estimatedRobotPose.estimatedPose).orElse(null);
+    }
+
+    @Override
+    public Pose3d getEstimatedPoseFieldOriented(Pose3d poseTargetOriented, int aprilId) {
+        Transform3d transform3d = new Transform3d(poseTargetOriented.getTranslation(), poseTargetOriented.getRotation());
+        return VisionConstants.TARGET_POSITION_REAL[aprilId].plus(transform3d);
+    }
+
+    // From Barel 😘:
+// code code code coding the code with some code to fix the code becasue i want code, code code code... all day long - just code!
+
+    public void setPipeLine(int pipeLineIndex) {
+        photonCamera.setPipelineIndex(pipeLineIndex);
+    }
+
+    @Override
+    public void updateInputs(VisionInputs inputs) {
+        inputs.targetID = target.getFiducialId();
+        inputs.hasTargets = result.hasTargets();
+        inputs.yaw = target.getYaw();
+        inputs.pitch = target.getPitch();
+        inputs.targetSkew = target.getSkew();
+        inputs.area = target.getArea();
+        inputs.latency = 0;
+        inputs.poseTargetOriented3d = getEstimatedPoseTargetOriented();
+        inputs.poseFieldOriented3d = getEstimatedPoseFieldOriented(inputs.poseTargetOriented3d, inputs.targetID);
+        inputs.poseTargetOriented = new double[]{
+                inputs.poseTargetOriented3d.getX(),
+                inputs.poseTargetOriented3d.getY(),
+                inputs.poseTargetOriented3d.getZ(),
+                inputs.poseTargetOriented3d.getRotation().getQuaternion().getX(),
+                inputs.poseTargetOriented3d.getRotation().getQuaternion().getY(),
+                inputs.poseTargetOriented3d.getRotation().getQuaternion().getZ(),
+                inputs.poseTargetOriented3d.getRotation().getQuaternion().getW()};
```

Comment:
If the PhotonVision API does indeed return target pose oriented, these variables would need to be renamed. And the `getEstimatedPoseTargetOriented()` method.